nucliadb/src/migrations/0024_assign_models_to_kbs.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (C) 2021 Bosutech XXI S.L.
+#
+# nucliadb is offered under the AGPL v3.0 and as commercial software.
+# For commercial licensing, contact us at info@nuclia.com.
+#
+# AGPL:
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+
+"""Migration #24
+
+Vectorsets are coming and we need to be ready at nucliadb. Vector index config
+shouldn't be stored anymore in the `Shards` protobuffer, we need to migrate to
+the new vectorsets config.
+
+This migration asks learning_config for each KB configuration and saves the
+model name as the vectorset_id. Creates a vectorset configuration for each model
+and deprecates the vectors index config from the `Shards` protobuffer.
+
+This migration should work for onprem and hosted deployments, as
+learning_proxy handles which API is used (internal or external)
+
+"""
+
+import logging
+
+from nucliadb import learning_proxy
+from nucliadb.common import datamanagers
+from nucliadb.migrator.context import ExecutionContext
+from nucliadb_protos import (
+    knowledgebox_pb2,
+    nodewriter_pb2,
+)
+
+logger = logging.getLogger(__name__)
+
+
+async def migrate(context: ExecutionContext) -> None: ...
+
+
+async def migrate_kb(context: ExecutionContext, kbid: str) -> None:
+    async with context.kv_driver.transaction(read_only=True) as txn:
+        vectorsets_count = len([vs async for vs in datamanagers.vectorsets.iter(txn, kbid=kbid)])
```

Comment:
```suggestion
        vectorsets_count = len(list(datamanagers.vectorsets.iter(txn, kbid=kbid)))
```

Reply:
Does this work? `iter` is an async iterator

Reply:
Ah yeah, maybe not then :sweat: